groups/script.js

```diff
@@ -178,25 +208,43 @@ searchInput.addEventListener('keyup', () => {
  * TO ASSIGN YOURSELF A ROLE
  */
 const buttonAddRole = document.querySelector('.btn-add-role');
-buttonAddRole.addEventListener('click', async function () {
+async function addrole() {
   if (memberAddRoleBody?.userid && memberAddRoleBody?.roleid !== '') {
     loader.classList.remove('hidden');
 
-    addGroupRoleToMember(memberAddRoleBody)
-      .then((res) => {
-        const groupNameElement = document.getElementById(
-          `name-${memberAddRoleBody.roleid}`,
-        );
-        const currentCount = groupNameElement.getAttribute('data-member-count');
-        if (currentCount !== null && currentCount !== undefined) {
-          groupNameElement.setAttribute('data-member-count', +currentCount + 1);
-        }
-        alert(res.message);
-      })
-      .catch((err) => alert(err.message))
-      .finally(() => loader.classList.add('hidden'));
+    try {
+      // Add the role to the member
+      const res = await addGroupRoleToMember(memberAddRoleBody);
+
+      const groupNameElement = document.getElementById(
+        `name-${memberAddRoleBody.roleid}`,
+      );
+      const currentCount = groupNameElement.getAttribute('data-member-count');
+      if (currentCount !== null && currentCount !== undefined) {
+        groupNameElement.setAttribute('data-member-count', +currentCount + 1);
+      }
+      alert(res.message);
+      if (isDev) {
+        // After adding the role, re-fetch the user group data to update it
+        UserGroupData = await getUserGroupRoles();
+
+        // Update the button state with the refreshed data
+        updateButtonState();
+      }
+    } catch (err) {
+      alert(err.message);
+    } finally {
+      loader.classList.add('hidden');
+    }
   }
-});
+}
+
+/**
+ * TO REMOVE YOURSELF OF A ROLE
+ */
+async function removeRoleHandler() {
+  console.log('Remove function to be added after this pr');
```

Comment:
Please add a TODO comment for upcoming features

Reply:
Done